### PR TITLE
Implement vite

### PR DESCRIPTION
**Pull request by ryanolee**
# What
This PR migrates from [create react app](https://create-react-app.dev/) to [vite](https://vitejs.dev/).

# Why
Using vite significantly reduces overall build times making the local dev experience better. This allows for 

# How
By removing CRA and adding Vite. Files containing jsx now use the ".jsx" extension over the standard ".js". The way environment variables are referenced has also changed

# Covers
- #32 

# Checklist
- [ ] Are the issues being addressed linked to this PR?
- [ ] Do all commit messages start with the issue number?
- [ ] Are all code changes sufficiently tested? (N/a)
- [ ] Are there screenshots for UI changes? (N/a)

### CODE REVIEW

File: src/firebase.config.js [hunk 1]
```diff
@@ -20,15 +20,16 @@ export const auth = getAuth(app)
 export const firestore = getFirestore(app)
 
 // Attach emulator in the event it has been enabled
+console.log([import.meta.env.DEV, import.meta.env.VITE_FIREBASE_EMULATOR_ENABLED])
```
Comment (LodenH16) [inline src/firebase.config.js]: Leftover console.log 😉

File: src/service-worker.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ registerRoute(
 
     return true;
   },
-  createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html')
+  createHandlerBoundToURL(process.env.VITE_PUBLIC_URL + '/index.html')
```
Comment (LodenH16) [inline src/service-worker.js]: Should this use the `import.meta.env` convention like the other env variables? I've never used Vite so I genuinely don't know😅

**Review by LodenH16 — CHANGES_REQUESTED** [review]
Everything looks great! My machine built in half a second, crazy fast 🔥!! Great idea to use Vite and great work!!

File: src/service-worker.js (hunk 2 above)
Comment (ryanolee) [reply src/service-worker.js]: Ahh yep that needs changing

File: src/firebase.config.js (hunk 1 above)
Comment (ryanolee) [reply src/firebase.config.js]: Whoops :P

**Pull request merged**